src/utils/hooks/useGetFullReviewStructure.tsx

```diff
@@ -1,115 +1,95 @@
 import { useState, useEffect } from 'react'
 import { FullStructure } from '../types'
 import {
-  ApplicationResponse,
   Review,
   ReviewResponse,
   useGetReviewNewQuery,
   ReviewQuestionAssignment,
-  ReviewStatus,
 } from '../generated/graphql'
-import addEvaluatedResponsesToStructure from '../helpers/structure/addEvaluatedResponsesToStructure'
 import { useUserState } from '../../contexts/UserState'
 import addThisReviewResponses from '../helpers/structure/addThisReviewResponses'
 import addElementsById from '../helpers/structure/addElementsById'
 import generateReviewProgress from '../helpers/structure/generateReviewProgress'
 
 interface useGetFullReviewStructureProps {
-  structure: FullStructure
+  fullApplicationStructure: FullStructure
   reviewAssignmentId: number
+  filteredSectionIds?: number[]
 }
 
 const useGetFullReviewStructure = ({
-  structure,
+  fullApplicationStructure,
   reviewAssignmentId,
+  filteredSectionIds,
 }: useGetFullReviewStructureProps) => {
-  const [fullStructure, setFullStructure] = useState<FullStructure>()
+  const [fullReviewStructure, setFullReviewStructure] = useState<FullStructure>()
   const [structureError, setStructureError] = useState('')
 
   const {
     userState: { currentUser },
   } = useUserState()
 
+  // filter by all sections if sections are not supplied
+  const sectionIds =
+    filteredSectionIds ||
+    fullApplicationStructure?.sortedSections?.map((section) => section.details.id) ||
+    []
+
+  console.log({ sectionIds })
+
   const { data, error } = useGetReviewNewQuery({
     variables: {
       reviewAssignmentId,
       userId: currentUser?.userId as number,
+      sectionIds,
     },
   })
 
   useEffect(() => {
+    console.log(fullApplicationStructure)
     if (error) return
 
     if (!data) return
 
     const reviewAssignment = data.reviewAssignment
-    if (!reviewAssignment) return setStructureError('Cannot find review Assignemnt')
-
-    addEvaluatedResponsesToStructure({
-      structure,
-      applicationResponses: reviewAssignment.application?.applicationResponses
-        ?.nodes as ApplicationResponse[],
-      currentUser,
-      evaluationOptions: {
-        isEditable: false,
-        isVisible: true,
-        isRequired: false,
-        isValid: false,
-      },
-    }).then((evaluatedStructure: FullStructure) => {
```

Comment:
Are responses evaluated before that?

Reply:
yep now in useGetFullApplicationStructure (called before any useGetFullReviewStructure)